dxr/static/js/code-highlighter.js

```diff
@@ -115,21 +115,25 @@ $(function () {
         }
     }
 
-    //update places where hash location is used: window, permalink
+    //update places where hash location is used: window, sync-links
     function updateHash(hash) {
-        if (permalink.length > 0)
-            updatePermalink(hash);
+        syncLinks.each(function() {
+            updateLink($(this), hash);
+        });
         history.replaceState(null, '', hash);
     }
 
-    //update the permalink href based on the windowHash.
-    function updatePermalink(windowHash) {
-        var permalink_href = permalink.attr('href'),
-            hash_loc = permalink_href.indexOf('#');
-        // If the link already has #, then cut that it off.
+    //update the link href based on the windowHash.
+    function updateLink(anchor, windowHash) {
+        var href = anchor.attr('href'),
+            hash_loc = href.indexOf('#');
+        // If the link already has #, then cut it off.
         if (hash_loc >= 0)
-            permalink_href = permalink_href.substring(0, hash_loc);
-        permalink.attr('href', permalink_href + windowHash);
+            href = href.substring(0, hash_loc);
+        // hg web uses #l and GitHub uses #L but supports #l, so we use #l here.
```

Comment:
I like the idea, but I worry that it's more complicated than we expect, since DXR also has multi-ranges highlighting, e.g. https://dxr.allizom.org/mozilla-central/source/moz.build#7-13,22,32-43